cmake/OmimConfig.cmake

```diff
@@ -6,3 +6,81 @@ set(OMIM_WARNING_FLAGS
 set(3PARTY_INCLUDE_DIRS "${OMIM_ROOT}/3party/boost")
 set(OMIM_DATA_DIR "${OMIM_ROOT}/data")
 set(OMIM_USER_RESOURCES_DIR "${OMIM_ROOT}/data")
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+# GCC 10.0 is required to support <charconv> header inclusion in base/string_utils.hpp
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10.0)
+  message(FATAL_ERROR "Minimum supported g++ version is 10.0, yours is ${CMAKE_CXX_COMPILER_VERSION}")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(PCH_EXTENSION "pch")
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set(PCH_EXTENSION "gch")
+endif()
+
+if (NJOBS)
+  message(STATUS "Number of parallel processes: ${NJOBS}")
+  set(CMAKE_JOB_POOLS custom=${NJOBS})
+  set(CMAKE_JOB_POOL_COMPILE custom)
+  set(CMAKE_JOB_POOL_LINK custom)
+  set(CMAKE_JOB_POOL_PRECOMPILE_HEADER custom)
+endif()
+
+if (USE_ASAN AND USE_TSAN)
+  message(FATAL_ERROR "Can't use two different sanitizers together")
+elseif (USE_ASAN)
+  message(STATUS "Address Sanitizer is enabled")
+  add_compile_options(
+      "-fsanitize=address"
+      "-fno-omit-frame-pointer"
+  )
+elseif (USE_TSAN)
+  message(STATUS "Thread Sanitizer is enabled")
+  add_compile_options(
+      "-fsanitize=thread"
+      "-fno-omit-frame-pointer"
+  )
+endif()
```

Comment:
Do they both work well with gcc and clang?

Reply:
In theory yes, they should
In practice (in our project) I have no idea